### ISSUE TITLE
panic when getting local description

### ISSUE THREAD

**Issue by maxmuen** [lerouxrgd/datachannel-rs#15]
Getting a panic, when tring to get the local_description. This problem appeared with version 0.5 and was not present in 0.4.
When migrating from 0.4 to 0.5, i changed `add_data_channel_ex` to `create_data_channel_ex` and set `disable_auto_negotiation=true`.

```rust
lazy_static! {
    static ref DATA_CHANNEL: DataChannelInit = DataChannelInit::default()
        .negotiated()
        .protocol("data")
        .manual_stream()
        .stream(1)
        .reliability(
            Reliability::default()
                .unordered()
                .unreliable()
                .max_retransmits(0),
        );
}

....

let data_channel = connection.create_data_channel_ex(
    "data",
    DataChannel { ... },
    &DATA_CHANNEL,
)?;
self.connection
  .set_remote_description(&SessionDescription {
      sdp_type: SdpType::Offer,
      sdp,
  })?;
let SessionDescription { mut sdp, sdp_type } = self
    .connection
    .local_description() // <- panic here
    .ok_or_else(|| anyhow!("no local description was generated"))?;
```

Stack Trace
```
thread 'tokio-runtime-worker' panicked at 'Couldn't get buffer size: NotAvailable', /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/datachannel-0.5.0/src/peerconnection.rs:486:14
stack backtrace:
   0: rust_begin_unwind
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/core/src/panicking.rs:92:14
   2: core::option::expect_none_failed
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/core/src/option.rs:1329:5
   3: core::result::Result<T,E>::expect
             at /home/max/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:997:23
   4: datachannel::peerconnection::RtcPeerConnection<P>::read_string_ffi
             at /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/datachannel-0.5.0/src/peerconnection.rs:485:24
   5: datachannel::peerconnection::RtcPeerConnection<P>::local_description
             at /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/datachannel-0.5.0/src/peerconnection.rs:385:19
```

**Comment by paullouisageneau** [comment]
Since you set `disable_auto_negotiation=true`, you need to call `set_local_description` after `set_remote_description` for the local description to be generated and retrievable with `local_description`.

**Comment by maxmuen** [comment]
calling `set_local_description` did fix the panic, though `local_description` should not panic in the first place.

**Comment by lerouxrgd** [comment]
@maxmuen  thank you for reporting this, indeed it shouldn't panic but the only alternative would be to return `None` I guess.

@paullouisageneau it boils down to [this line](https://github.com/lerouxrgd/datachannel-rs/blob/bc72a3b24824a452561834568d361e7394095920/src/peerconnection.rs#L485) which is the "calling with null ptr to get the buffer size" trick, is it normal that it doesn't succeed in this scenario ?

**Comment by paullouisageneau** [comment]
> @paullouisageneau it boils down to [this line](https://github.com/lerouxrgd/datachannel-rs/blob/bc72a3b24824a452561834568d361e7394095920/src/peerconnection.rs#L485) which is the "calling with null ptr to get the buffer size" trick, is it normal that it doesn't succeed in this scenario ?

@lerouxrgd Yes this is expected, if there is no local description set, `rtcGetLocalDescription()` returns `RTC_ERR_NOT_AVAIL`.

**Comment by lerouxrgd** [comment]
@maxmuen It has been fixed on master, could you give it a try ?

**Comment by maxmuen** [comment]
@lerouxrgd can confirm that None  is returned instead of a panic. But it appears the error `Couldn't get buffer size: NotAvailable` is being logged twice, for every `local_description()` call

**Comment by lerouxrgd** [comment]
This is strange, are you sure you aren't calling it twice somewhere ?

**Comment by lerouxrgd** [comment]
Yes `local_description` calls both `sys::rtcGetLocalDescription` and `sys::rtcGetLocalDescriptionType` and each of them will print the log error message.

**Comment by lerouxrgd** [comment]
Fixed in `0.5.1`.
